src/server/server.py

```diff
@@ -1809,9 +1848,18 @@ def on_shutdown_pi():
     trigger_event(Evt.SHUTDOWN)
     CLUSTER.emit('shutdown_pi')
     emit_priority_message(__('Server has shut down.'), True)
-    logger.info('Shutdown pi')
-    gevent.sleep(1);
-    os.system("sudo shutdown now")
+    logger.info('Performing system shutdown')
+    stop_background_threads()
+    gevent.sleep(0.5)
+    gevent.spawn(SOCKET_IO.stop)  # shut down flask http server
+    if RHUtils.isSysRaspberryPi():
+        gevent.sleep(0.1)
+        logger.debug("Executing system command:  sudo shutdown now")
+        log.wait_for_queue_empty()
+        log.close_logging()
+        os.system("sudo shutdown now")
+    else:
+        logger.debug("Not executing system shutdown command because not RPi")
```

Comment:
Since this should never happen, does it warrant a higher log level?

Reply:
Wouldn't hurt